primitives/src/pcs/univariate_kzg/mod.rs

```diff
@@ -378,17 +386,19 @@ where
 
         // Step 2. evaluate h(X) with coeffs \vec{h} at `points` using FFT
         let mut h_poly = GeneralDensePolynomial::from_coeff_vec(h_vec);
-        // NOTE!! this is the trickist subtle line in this function !!
-        // `polynomial` has a degree `d` which is guaranteed to be power-of-two to use
-        // Toeplitz fast mul algorithm. h(X) has a degree `d-1` which cause a plain
-        // domain instantiation on `h_poly` half the size of that for `polynomial`,
-        // which leads to different vector of roots of unity used, ultimately leading to
-        // wrong proofs. Thus, to make sure during FFT, the same vector of roots of
-        // unity are used, we need to manually resize the coefficients so that
-        // `GeneralDensePolynomial::batch_evaluate` are informed to use the domain of
-        // size (d+1).next_power_of_two().
-        h_poly.coeffs.resize(padded_len, E::G1::zero());
 
+        if padded_len == poly_coeffs.len() {
```

Comment:
because the tricky degree resizing is only relevant when `d=poly_coeffs.len()` is a power of two. in other cases, we don't need the padding (in another word, comment this condition out, you will fail the test)